crates/bevy_ecs/src/query/iter.rs

```diff
@@ -481,6 +485,8 @@ where
 struct QueryIterationCursor<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> {
     table_id_iter: std::slice::Iter<'s, TableId>,
     archetype_id_iter: std::slice::Iter<'s, ArchetypeId>,
+    entities: &'w [Entity],
+    archetype_entities: &'w [ArchetypeEntity],
```

Comment:
I think `table_entities` instead of `entities` would be a bit clearer, more than that it'd be good to express that its like ``Either<&'w [Entity] /* from a `Table` used with dense queries */, &'w [ArchetypeEntity] /* from a `Archetype` used with non dense queries */>``. I think not necessarily worth blocking this PR on though, it'd be pretty easy for me to do this once merged

Reply:
Was going to follow up this one with the union PR, which should allow us to do the same with in this case.

Will rename this when I can.